schemachanger: v22.1.5: deleting ambiguously created lease: encountered poisoned latch

This issue was autofiled by Sentry. It represents a crash or reported error on a live cluster with telemetry enabled.

Sentry link: https://sentry.io/organizations/cockroach-labs/issues/3480501595/?referrer=webhooks_plugin

Panic message:

>replica_circuit_breaker.go:294: error executing ×: ×: deleting ambiguously created lease: ×: replica unavailable: (n6,s6):7 unable to serve request to r7:× [(n4,s4):4, (n3,s3):2, (n2,s2):3, (n6,s6):7, (n1,s1):6, next=8, gen=16]: lost quorum (down: (n3,s3):2,(n2,s2):3,(n1,s1):6); closed timestamp: 1659689237.890263121,0 (2022-08-05 08:47:17); raft status: {"id":"7","term":10,"vote":"7","commit":5598,"lead":"7","raftState":"StateLeader","applied":5598,"progress":{"6":{"match":0,"next":5414,"state":"StateProbe"},"7":{"match":5642,"next":5643,"state":"StateReplicate"},"2":{"match":0,"next":5414,"state":"StateProbe"},"3":{"match":5598,"next":5599,"state":"StateProbe"},"4":{"match":5642,"next":5643,"state":"StateReplicate"}},"leadtransferee":"0"}: encountered poisoned latch ×@1659689241.013506924,0
(1) attached stack trace
  -- stack trace:
  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scrun.executeStage
  | 	github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scrun/scrun.go:160
  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scrun.runTransactionPhase
  | 	github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scrun/scrun.go:73
  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scrun.RunPreCommitPhase
  | 	github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scrun/scrun.go:47
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).runPreCommitStages
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:3201
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).commitSQLTransactionInternal
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:947
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).commitSQLTransaction
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:888
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).handleAutoCommit
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:2074
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState.func6
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:520
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:734
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:143
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func1
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1884
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1888
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1810
  | github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:756
  | github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1
  | 	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:727
Wraps: (2) error executing ×
Wraps: (3) assertion failure
Wraps: (4) ×
Wraps: (5) ×
Wraps: (6) ×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
×
Wraps: (7) attached stack trace
  -- stack trace:
  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).execInternal.func1.1
  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:685
  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next.func1
  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:282
  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next
  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:333
  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).ExecEx
  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:542
  | github.com/cockroachdb/cockroach/pkg/sql/descmetadata.metadataUpdater.DeleteDescriptorComment
  | 	github.com/cockroachdb/cockroach/pkg/sql/descmetadata/metadata_updater.go:60
  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scexec.updateDescriptorMetadata
  | 	github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scexec/exec_mutation.go:246
  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scexec.executeDescriptorMutationOps
  | 	github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scexec/exec_mutation.go:71
  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scexec.ExecuteStage
  | 	github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scexec/executor.go:32
  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scrun.executeStage
  | 	github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scrun/scrun.go:153
  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scrun.runTransactionPhase
  | 	github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scrun/scrun.go:73
  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scrun.RunPreCommitPhase
  | 	github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scrun/scrun.go:47
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).runPreCommitStages
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:3201
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).commitSQLTransactionInternal
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:947
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).commitSQLTransaction
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:888
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).handleAutoCommit
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:2074
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState.func6
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:520
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:734
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:143
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func1
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1884
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1888
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1810
  | github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:756
  | github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1
  | 	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:727
Wraps: (8) ×
Wraps: (9) attached stack trace
  -- stack trace:
  | github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.storage.acquire.func1
  | 	github.com/cockroachdb/cockroach/pkg/sql/catalog/lease/storage.go:120
  | [...repeated from below...]
Wraps: (10) deleting ambiguously created lease
Wraps: (11) attached stack trace
  -- stack trace:
  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).execInternal.func1.1
  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:685
  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next.func1
  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:282
  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next
  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:333
  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).ExecEx
  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:542
  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).Exec
...


<details>
<summary>Stacktrace (expand for inline code snippets):</summary>

https://github.com/cockroachdb/cockroach/blob/a30a663cbd9323d34d50f343dd038af64671e25f/pkg/sql/schemachanger/scrun/scrun.go#L159-L161 in pkg/sql/schemachanger/scrun.executeStage
https://github.com/cockroachdb/cockroach/blob/a30a663cbd9323d34d50f343dd038af64671e25f/pkg/sql/schemachanger/scrun/scrun.go#L72-L74 in pkg/sql/schemachanger/scrun.runTransactionPhase
https://github.com/cockroachdb/cockroach/blob/a30a663cbd9323d34d50f343dd038af64671e25f/pkg/sql/schemachanger/scrun/scrun.go#L46-L48 in pkg/sql/schemachanger/scrun.RunPreCommitPhase
https://github.com/cockroachdb/cockroach/blob/a30a663cbd9323d34d50f343dd038af64671e25f/pkg/sql/conn_executor.go#L3200-L3202 in pkg/sql.(*connExecutor).runPreCommitStages
https://github.com/cockroachdb/cockroach/blob/a30a663cbd9323d34d50f343dd038af64671e25f/pkg/sql/conn_executor_exec.go#L946-L948 in pkg/sql.(*connExecutor).commitSQLTransactionInternal
https://github.com/cockroachdb/cockroach/blob/a30a663cbd9323d34d50f343dd038af64671e25f/pkg/sql/conn_executor_exec.go#L887-L889 in pkg/sql.(*connExecutor).commitSQLTransaction
https://github.com/cockroachdb/cockroach/blob/a30a663cbd9323d34d50f343dd038af64671e25f/pkg/sql/conn_executor_exec.go#L2073-L2075 in pkg/sql.(*connExecutor).handleAutoCommit
https://github.com/cockroachdb/cockroach/blob/a30a663cbd9323d34d50f343dd038af64671e25f/pkg/sql/conn_executor_exec.go#L519-L521 in pkg/sql.(*connExecutor).execStmtInOpenState.func6
https://github.com/cockroachdb/cockroach/blob/a30a663cbd9323d34d50f343dd038af64671e25f/pkg/sql/conn_executor_exec.go#L733-L735 in pkg/sql.(*connExecutor).execStmtInOpenState
https://github.com/cockroachdb/cockroach/blob/a30a663cbd9323d34d50f343dd038af64671e25f/pkg/sql/conn_executor_exec.go#L142-L144 in pkg/sql.(*connExecutor).execStmt
https://github.com/cockroachdb/cockroach/blob/a30a663cbd9323d34d50f343dd038af64671e25f/pkg/sql/conn_executor.go#L1883-L1885 in pkg/sql.(*connExecutor).execCmd.func1
https://github.com/cockroachdb/cockroach/blob/a30a663cbd9323d34d50f343dd038af64671e25f/pkg/sql/conn_executor.go#L1887-L1889 in pkg/sql.(*connExecutor).execCmd
https://github.com/cockroachdb/cockroach/blob/a30a663cbd9323d34d50f343dd038af64671e25f/pkg/sql/conn_executor.go#L1809-L1811 in pkg/sql.(*connExecutor).run
https://github.com/cockroachdb/cockroach/blob/a30a663cbd9323d34d50f343dd038af64671e25f/pkg/sql/conn_executor.go#L755-L757 in pkg/sql.(*Server).ServeConn
https://github.com/cockroachdb/cockroach/blob/a30a663cbd9323d34d50f343dd038af64671e25f/pkg/sql/pgwire/conn.go#L726-L728 in pkg/sql/pgwire.(*conn).processCommandsAsync.func1
GOROOT/src/runtime/asm_amd64.s#L1580-L1582 in runtime.goexit
</details>

```
pkg/sql/schemachanger/scrun/scrun.go in pkg/sql/schemachanger/scrun.executeStage at line 160
pkg/sql/schemachanger/scrun/scrun.go in pkg/sql/schemachanger/scrun.runTransactionPhase at line 73
pkg/sql/schemachanger/scrun/scrun.go in pkg/sql/schemachanger/scrun.RunPreCommitPhase at line 47
pkg/sql/conn_executor.go in pkg/sql.(*connExecutor).runPreCommitStages at line 3201
pkg/sql/conn_executor_exec.go in pkg/sql.(*connExecutor).commitSQLTransactionInternal at line 947
pkg/sql/conn_executor_exec.go in pkg/sql.(*connExecutor).commitSQLTransaction at line 888
pkg/sql/conn_executor_exec.go in pkg/sql.(*connExecutor).handleAutoCommit at line 2074
pkg/sql/conn_executor_exec.go in pkg/sql.(*connExecutor).execStmtInOpenState.func6 at line 520
pkg/sql/conn_executor_exec.go in pkg/sql.(*connExecutor).execStmtInOpenState at line 734
pkg/sql/conn_executor_exec.go in pkg/sql.(*connExecutor).execStmt at line 143
pkg/sql/conn_executor.go in pkg/sql.(*connExecutor).execCmd.func1 at line 1884
pkg/sql/conn_executor.go in pkg/sql.(*connExecutor).execCmd at line 1888
pkg/sql/conn_executor.go in pkg/sql.(*connExecutor).run at line 1810
pkg/sql/conn_executor.go in pkg/sql.(*Server).ServeConn at line 756
pkg/sql/pgwire/conn.go in pkg/sql/pgwire.(*conn).processCommandsAsync.func1 at line 727
GOROOT/src/runtime/asm_amd64.s in runtime.goexit at line 1581
```

| Tag | Value |
|---|---|
| Cockroach Release | `v22.1.5` |
| Cockroach SHA: | a30a663cbd9323d34d50f343dd038af64671e25f |
| Platform | linux amd64 |
| Distribution | CCL |
| Environment | v22.1.5 |
| Command | server |
| Go Version | ``|
| # of CPUs ||
| # of Goroutines ||


Jira issue: CRDB-18364

--- a/pkg/sql/schemachanger/scplan/plan.go
+++ b/pkg/sql/schemachanger/scplan/plan.go
@@ -95,6 +95,7 @@ func makePlan(p *Plan) (err error) {
 			}
 			err = rAsErr
 		}
+		err = errors.WithAssertionFailure(err)
 	}()
 	{
 		start := timeutil.Now()

--- a/pkg/sql/schemachanger/scplan/plan_explain.go
+++ b/pkg/sql/schemachanger/scplan/plan_explain.go
@@ -41,7 +41,7 @@ func (p Plan) DecorateErrorWithPlanDetails(err error) (retErr error) {
 	if p.Graph != nil {
 		err = addDetail(err, "dependencies graphviz URL", "dependencies graphviz: ", p.DependenciesURL)
 	}
-	return errors.WithAssertionFailure(err)
+	return err
 }
 
 func addDetail(

